modopt/opt/proximity.py

```diff
@@ -688,3 +689,353 @@ def _cost_method(self, *args, **kwargs):
             print(' - ELASTIC NET (X):', cost_val)
 
         return cost_val
+
+
+class KSupportNorm(ProximityParent):
+    r"""K-support norm proximity operator
+
+    This class defines the squarred K-support norm proximity operator
+    described in [M2016].
+
+    \underset{y \in \mathbb{C}^{N}}{\text{min}} 0.5*\|x - y|\_2^2 +
+    \frac{\beta}{2}\text{min}\{\sum_{I \in \mathcal{G}_k}\|v_I\|_2^2:
+    \text{supp} (v_I) \subseteq I, \sum_{I \in \mathcal{G}_k} v_I = y\}
+
+    where G_k is the collection of all subset containing at most k elements.
+
+    Parameters
+    ----------
+    thresh : float
+        Threshold value
+    k_value : int
+        Hyper-parameter of the k-support norm, equivalent to the cardinality
+        value for the overlapping group lasso. k should included in
+        {1, ..., dim(input_vector)}
+
+    Notes:
+    ------
+    The k-support norm can be seen as an extension to the group-LASSO with
+    overlaps with groups of cardianlity at most equal to k.
+    When k = 1 the norm is equivalent to the L1-norm.
+    When k = dimension of the input vector than the norm is equivalent to the
+    L2-norm.
+    The dual of this norm correspond to the sum of the k biggest input entries
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from modopt.opt.proximity import KSupportNorm
+    >>> A = np.arange(5)*5
+    array([ 0,  5, 10, 15, 20])
+    >>> prox_op = KSupportNorm(beta=3, k_value=1)
+    >>> prox_op.op(A)
+    array([ 0.,  0.,  0., 0., 5.])
+    >>> prox_op.cost(A, verbose=True)
+     - OWL NORM (X): 7500.0
+    7500.0
+    """
+
+    def __init__(self, beta, k_value):
+        self.beta = beta
+        self.k_value = k_value
+        if self.k_value < 1:
+            raise ValueError("The k parameter should be greater or "
+                             "equal than 1")
```

Comment:
This is fine unless you expect the user to interact with `k_value` in which case it would be better to define an explicit getter and setter using `@property` and `@k_value.setter`.

Reply:
I'm adding this in just in case the parameters have been changed by the user